src/pykeen/models/base.py

```diff
@@ -1195,3 +1197,174 @@ def _can_slice(fn) -> bool:
 
 class MultimodalModel(EntityRelationEmbeddingModel):
     """A multimodal KGE model."""
+
+
+class InteractionFunction(nn.Module):
+    """Base class for interaction functions."""
+
+    def forward(
+        self,
+        h: torch.FloatTensor,
+        r: torch.FloatTensor,
+        t: torch.FloatTensor,
+    ) -> torch.FloatTensor:
+        """Score the given triples.
+
+        :param h: shape: (batch_size, num_heads, d_e)
+            The head representations.
+        :param r: shape: (batch_size, num_relations, d_r)
+            The relation representations.
+        :param t: shape: (batch_size, num_tails, d_e)
+            The tail representations.
+
+        :return: shape: (batch_size, num_heads, num_relations, num_tails)
+            The scores.
+        """
+        raise NotImplementedError
+
+    def score_t(
```

Comment:
why only `score_t` and not `score_r` and `score_h`?

Reply:
It should also have score_r / score_h, but I wanted to await feedback on score_t before doing the other two.